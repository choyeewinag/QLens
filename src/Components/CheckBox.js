import React, { useState, useEffect } from 'react';
// import MongoSchemaIDE from 'MongoSchemaIDE';

const CheckBox = (props) => {
  return (
    <div className="checkboxContainer" id="box">
      <form className="checkboxForm">
        <label className="checkboxLabel">
          <input type="checkbox" name={props.name} onClick={props.clicked} />
          {/* name is the key which is the obj */}
          {props.name}
        </label>
      </form>
    </div>
  )
}



export default CheckBox;