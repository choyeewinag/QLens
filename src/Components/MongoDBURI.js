import React, { useState, useEffect } from "react"; // userEffect to fetch instead of component mount
import '../public/index.css';

// input field for the mongoDB uri
const MongoDBURI = ({schemaData, uriData, submitbtn, geturi}) =>  {
    return (
        <div className="formContainer">
            <div className="formClass">
              <form className="form__group field">
                  <input className="form__field" type="input" onChange= {geturi} id='name' placeholder=" Input Your MongoDB URI"/>
                  <label for="name" className="form__label"> MongoDB URI </label>
                  <input className="URISubmitButton" type="submit" value="Submit" onClick={submitbtn} style={{display: "none"}}/>
              </form>
            </div>
        </div>
    )
}

export default MongoDBURI;
