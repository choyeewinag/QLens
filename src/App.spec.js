//https://www.robinwieruch.de/react-testing-jest
//https://jestjs.io/docs/en/snapshot-testing

  import React from 'react';
  import renderer from 'react-test-renderer';
  import CheckBox from '../src/Components/CheckBox';
  import MongoDBURI from '../src/Components/MongoDBURI'; // had to comment out public css file
//   import DropDownMenu from '../src/Components/DropDownMenu'; // had to comment out public css file
//   import MongoSchemaIDE from '../src/Components/MongoSchemaIDE'; // had to comment out public css file

  describe('My Test Suite', () => {
    it('My Test Case', () => {
      expect(true).toEqual(true);
    });
  });
   
  describe('CheckBox Component Renders Correctly', () => {
    test('snapshot of CheckBox Component renders', () => {
      const component = renderer.create(<CheckBox/>);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('MongoDBURI Component Renders Correctly', () => {
    test('snapshot of MongoDBURI Component renders', () => {
      const component = renderer.create(<MongoDBURI/>);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

//   describe('DropDownMenu Component Renders Correctly', () => {
//     test('snapshot of DropDownMenu Component renders', () => {
//       const component = renderer.create(<DropDownMenu/>);
//       let tree = component.toJSON();
//       expect(tree).toMatchSnapshot();
//     });
//   });

//   describe('MongoSchemaIDE Component Renders Correctly', () => {
//     test('snapshot of MongoSchemaIDE Component renders', () => {
//       const component = renderer.create(<MongoSchemaIDE/>);
//       let tree = component.toJSON();
//       expect(tree).toMatchSnapshot();
//     });
//   });