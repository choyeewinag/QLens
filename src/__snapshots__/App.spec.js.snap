// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component Renders Correctly snapshot of CheckBox Component renders 1`] = `
<div
  className="checkboxContainer"
  id="box"
>
  <form
    className="checkboxForm"
  >
    <label
      className="checkboxLabel"
    >
      <input
        type="checkbox"
      />
    </label>
  </form>
</div>
`;

exports[`Counter snapshot renders 1`] = `
<div
  className="checkboxContainer"
  id="box"
>
  <form
    className="checkboxForm"
  >
    <label
      className="checkboxLabel"
    >
      <input
        type="checkbox"
      />
    </label>
  </form>
</div>
`;

exports[`MongoDBURI Component Renders Correctly snapshot of MongoDBURI Component renders 1`] = `
<div
  className="formContainer"
>
  <div
    className="formClass"
  >
    <form
      className="form__group field"
    >
      <input
        className="form__field"
        id="name"
        placeholder=" Input Your MongoDB URI"
        type="input"
      />
      <label
        className="form__label"
        for="name"
      >
         MongoDB URI 
      </label>
      <input
        className="URISubmitButton"
        style={
          Object {
            "display": "none",
          }
        }
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
